### PR TITLE
Solved issue #4187- Tower Defence Game

### LINKED ISSUE #4187: [New game]: Tower defence
### 🎮 Game Request

Game logic and basic description
In this game, players place defensive towers along a path to stop waves of enemies from reaching the end. Each tower has different abilities and ranges, and players can upgrade towers to improve their effectiveness.

### Point down the features

Game points
Multiple Tower Types:

Basic Tower: Standard tower with medium range and attack speed.
Sniper Tower: Long-range tower with high damage but slow attack speed.
Rapid Fire Tower: Short-range tower with low damage but very fast attack speed.
Splash Damage Tower: Deals area-of-effect damage to multiple enemies at once.
Enemy Variety:

Basic Enemy: Standard enemy with average speed and health.
Fast Enemy: High-speed enemy with lower health.
Tank Enemy: Slow-moving enemy with high health.
Flying Enemy: Can only be attacked by certain towers.

### Select program in which you are contributing

GSSoC24

### Code of Conduct

- [X] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) of this project.

### PR DESCRIPTION
**Pull request by Aditi22Bansal**
## PR Description 📜

Please include summary related to the issue you have fixed and describe your PR in brief over here by specifying the issue number on which you were working below

In this game, players place defensive towers along a path to stop waves of enemies from reaching the end. Each tower has different abilities and ranges, and players can upgrade towers to improve their effectiveness.

Fixes # <your_issue_number>
Fixes #4187 

<hr>
 
## Mark the task you have completed ✅

- [x] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) & [CODE OF CONDUCT](https://github.com/kunjgit/GameZone/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I have followed proper naming convention showed in [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md)
- [x] I have added screenshot for website preview in assets/images 
- [x] I have added entries for my game in main README.md
- [x] I have added README.md in my folder 
- [x] I have added working video of the game in README.md (optional)
- [x] I have specified the respective issue number for which I have requested the new game.

<hr>

## Add your screenshots(Optional) 📸
Screenshot 2024-06-04 004952
Screenshot 2024-06-04 005003
Tower_Game

--- 
<br>

## Thank you soo much for contributing to our repository 💗

### CODE REVIEW

**Comment by kunjgit** [comment]
Thank you @Aditi22Bansal ,for creating the PR and contributing to our GameZone 💗 
 Review team will review the PR and will reach out to you soon! 😇 
 Make sure that you have marked all the tasks that you are done with ✅. 
 Thank you for your patience! 😀

**Review by kunjgit — CHANGES_REQUESTED** [review]
Hello @Aditi22Bansal,
You need to make the following changes:
  - Please add an image to the assets/images directory.

Hoping that you will make those changes soon 🚀

**Comment by Aditi22Bansal** [comment]
@ishita-43 
Kindly review this and approve
I have done it according to the changes you said in the previous PR
If any changes are still left to be done
Kindly tell
Thanks

**Comment by Aditi22Bansal** [comment]
@ishita-43 
Kindly review the PR

**Comment by Aditi22Bansal** [comment]
@ishita-43 
@kunjgit 
Kindly review the PR

File: README.md [hunk 1]
```diff
@@ -335,12 +335,17 @@ This repository also provides one such platforms where contributers come over an
 | [Dsa_quiz_game](https://github.com/kunjgit/GameZone/tree/main/Games/Dsa_quiz_game) |
 | [Gravity_Simulation_Game](https://github.com/kunjgit/GameZone/tree/main/Games/Gravity_Simulation_Game) |
 | [Anagarm-Word-Game](https://github.com/kunjgit/GameZone/tree/main/Games/Anagarm-Word-Game) |
+| [Brick Buster Game](https://github.com/kunjgit/GameZone/tree/main/Games/Brick Buster) |
```
Comment (ishita-43) [inline README.md:338]: delete the duplicate copies before below brick buster game

**Review by ishita-43 — CHANGES_REQUESTED** [review]

**Comment by Aditi22Bansal** [comment]
image
@ishita-43 
Done kindly review

File: README.md [hunk 2]
```diff
@@ -340,11 +340,9 @@ This repository also provides one such platforms where contributers come over an
 |[Penguins Can't Fly](https://github.com/Will2Jacks/GameZoneForked/tree/Task/Games/Penguins_Can't_Fly)|
 | [Taash_Game](https://github.com/kunjgit/GameZone/tree/main/Games/Taash_Game)|
 | [Brick Buster Game](https://github.com/kunjgit/GameZone/tree/main/Games/Brick%20Buster) |
-| [Rapid_click_frenzy](https://github.com/kunjgit/GameZone/tree/main/Games/Rapid_click_frenzy) |
-| [Penguins Can't Fly](https://github.com/Will2Jacks/GameZoneForked/tree/Task/Games/Penguins_Can't_Fly) |
 | [Intellect Quest](https://github.com/Will2Jacks/GameZoneForked/tree/Task/Games/Intellect_Quest) |
-| [Taash_Game](https://github.com/kunjgit/GameZone/tree/main/Games/Taash_Game) |
 | [Number_Guessing_Game](https://github.com/kunjgit/GameZone/tree/main/Games/Number_Guessing_Game) |
+| [Modulo_Game](https://github.com/kunjgit/GameZone/tree/main/Games/Modulo_Game) |
```
Comment (ishita-43) [inline README.md:345]: remove this if not in  the repository

File: README.md [hunk 3]
```diff
@@ -340,11 +340,9 @@ This repository also provides one such platforms where contributers come over an
...
 |[Tower Defence Game](https://github.com/Will2Jacks/GameZoneForked/tree/Task/Games/Tower_Defence_Game)|
```
Comment (ishita-43) [inline README.md:346]: add penguin and frenzy in the end and make sure to place the penguin in the below

**Review by ishita-43 — CHANGES_REQUESTED** [review]

**Comment by Aditi22Bansal** [comment]
image
@ishita-43 
Kindly check now

File: README.md [hunk 4]
```diff
@@ -331,17 +331,18 @@ This repository also provides one such platforms where contributers come over an
...
+| [Rapid_click_frenzy](https://github.com/kunjgit/GameZone/tree/main/Games/Rapid_click_frenzy) 
+| [Taash_Game](https://github.com/kunjgit/GameZone/tree/main/Games/Taash_Game)|
 | [Brick Buster Game](https://github.com/kunjgit/GameZone/tree/main/Games/Brick%20Buster) |
-| [Rapid_click_frenzy](https://github.com/kunjgit/GameZone/tree/main/Games/Rapid_click_frenzy) |
-| [Penguins Can't Fly](https://github.com/Will2Jacks/GameZoneForked/tree/Task/Games/Penguins_Can't_Fly) |
 | [Intellect Quest](https://github.com/Will2Jacks/GameZoneForked/tree/Task/Games/Intellect_Quest) |
-| [Taash_Game](https://github.com/kunjgit/GameZone/tree/main/Games/Taash_Game) |
 | [Number_Guessing_Game](https://github.com/kunjgit/GameZone/tree/main/Games/Number_Guessing_Game) |
-| [Modulo_Game](https://github.com/kunjgit/GameZone/tree/main/Games/Modulo_Game) |
+| [Rapid_click_frenzy](https://github.com/kunjgit/GameZone/tree/main/Games/Rapid_click_frenzy) |
```
Comment (ishita-43) [inline README.md:343]: replace this one with modulo game

**Comment by Aditi22Bansal** [comment]
image
@ishita-43 
Now check

**Review by ishita-43 — APPROVED** [review]
Good to go!!

**Comment by Aditi22Bansal** [comment]
Thankyou so much @ishita-43 
Kindly merge

**Comment by ishita-43** [comment]
Merge will be done by Admin within 2-3 days. Till then you can explore other projects also and start working on issues.

**Comment by Aditi22Bansal** [comment]
Okay thanks

**Comment by kunjgit** [comment]
@Aditi22Bansal resolve merge conflicts

**Comment by ishita-43** [comment]
@kunjgit resolved!!

**Comment by Aditi22Bansal** [comment]
Yes @kunjgit 
The conflicts are already resolved

**Comment by Aditi22Bansal** [comment]
@ishita-43 
@kunjgit 
Kindly review and merge
I want to work on the other assigned issue
I have already worked just have to make the PR
It has been many days assigned to me that issue

**Comment by kunjgit** [comment]
@Aditi22Bansal resolve merge conflicts !

**Comment by Aditi22Bansal** [comment]
@kunjgit 
Kindly review 
Done the required changes

**Comment by Aditi22Bansal** [comment]
@kunjgit
Kindly review
Done the required changes
I want to work on the other issue

**Comment by ishita-43** [comment]
it will soon be merged by the admin!

**Comment by kunjgit** [comment]
> @Aditi22Bansal resolve merge conflicts !
> 

@Aditi22Bansal it's still there

**Comment by Aditi22Bansal** [comment]
@kunjgit 
Due to late in merging 
Conflicts are arising again and again 
I have resolved the conflicts again
But kindly tell how to make it a permanent change

**Comment by kunjgit** [comment]
Thank you  @Aditi22Bansal , for your valuable time and contribution in our GameZone 💗. 
 It’s our GameZone, so Let’s build this GameZone altogether !!🤝
  Hoping to see you soon with another PR again 😇 
 Wishing you all the best for your journey into Open Source🚀

**Pull request merged**